src/test/test-mountpoint-util.c

```diff
@@ -354,9 +354,9 @@ TEST(fstype_can_discard) {
 }
 
 TEST(fstype_can_norecovery) {
-        assert_se(fstype_can_norecovery("ext4"));
-        assert_se(!fstype_can_norecovery("vfat"));
-        assert_se(!fstype_can_norecovery("tmpfs"));
+        ASSERT_STREQ(fstype_norecovery_option("ext4"), "norecovery");
+        ASSERT_NULL(fstype_norecovery_option("vfat"));
+        ASSERT_NULL(fstype_norecovery_option("tmpfs"));
```

Comment:
Maybe better to add `assert_se(STR_IN_SET(fstype_norecovery_option("btrfs"), ...));` ?

Reply:
It depends a bit too much on the kernel I think, mount API needs to be properly supported, commits need to not be backported and such. I'd prefer if we leave this out.

Reply:
I mean `assert_se(STR_IN_SET(fstype_norecovery_option("btrfs"), "norecovery", "rescue=nologreplay"));`. You mean it does not work for some kernel versions?

Reply:
Ah, `mount_option_supported()` does not work for really old kernel...
Then, how about to assume "norecovery" works (without checking with `mount_option_supported()`) for "btrfs" when "rescue=nologreplay" does not work?

Reply:
> You can call it API breakage as you want but note that it's been deprecated for 3 years (in 5.11). Nobody asked for keeping the option so it got removed.

Or, then, return NULL instead of checking "norecovery" if the new option is not supported?

Reply:
The new mount API might have been backported to older kernels, we can't really make assumptions here. It's not impossible that there's a kernel out there with the new mount API but without norecovery or with the new mount API and with norecovery, so I'd prefer to not make such assumptions.